client/storybook/src/RedesignToggleDecorator.tsx

```diff
@@ -0,0 +1,48 @@
+import { DecoratorFunction } from '@storybook/addons'
+import isChromatic from 'chromatic/isChromatic'
+import React, { ReactElement, useEffect } from 'react'
+
+import { useRedesignToggle, REDESIGN_CLASS_NAME } from '@sourcegraph/shared/src/util/useRedesignToggle'
+
+const toggleRedesignClass = (element: HTMLElement, isRedesignEnabled: boolean): void => {
+    element.classList.toggle(REDESIGN_CLASS_NAME, isRedesignEnabled)
+}
+
+const updatePreview = (isRedesignEnabled: boolean): void => {
+    const iframe = document.querySelector('#storybook-preview-iframe') as HTMLIFrameElement | undefined
+
+    const iframeDocument = iframe?.contentDocument || iframe?.contentWindow?.document
+    const body = iframeDocument?.body
+
+    if (body) {
+        toggleRedesignClass(body, isRedesignEnabled)
+    }
+}
+
+const updateManager = (isRedesignEnabled: boolean): void => {
+    const manager = document.querySelector('body')
+
+    if (manager) {
+        toggleRedesignClass(manager, isRedesignEnabled)
+    }
+}
+
+export const RedesignToggleDecorator: DecoratorFunction<ReactElement> = (Story, context) => {
```

Comment:
```suggestion
export const redesignToggleDecorator: DecoratorFunction<ReactElement> = (Story, context) => {
```

nit: looks like Storybook decorators are named in camelCase?

Reply:
I've noticed this too and tried to use the lower case approach at first, but it causes the error in `useRedesignToggle` usage because it violates `react/rules-of-hooks`. I think keeping it this way is better than disabling the ESLint rule for a specific line.